From 96388566f2c31e46d2912f9205fed6271a7b3f93 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Sun, 28 Nov 2010 15:31:02 +0000
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: 602606a472963a67b234e6b5c99293de4aa9d06b
Subject: [PATCH 1903/2588] drm/i915/execbuffer: On error, starting unwinding
 from the previous object

As the error occurred on the current object, it means that its state was
not changed and so it should be excluded from the unwind.

Reported-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 602606a472963a67b234e6b5c99293de4aa9d06b)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem_execbuffer.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_execbuffer.c b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
index d540701..66c898c 100644
--- a/drivers/gpu/drm/i915/i915_gem_execbuffer.c
+++ b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
@@ -530,6 +530,9 @@ i915_gem_execbuffer_reserve(struct drm_device *dev,
 	} while (1);
 
 err:
+	obj = list_entry(obj->exec_list.prev,
+			 struct drm_i915_gem_object,
+			 exec_list);
 	while (objects != &obj->exec_list) {
 		if (obj->gtt_space)
 			i915_gem_object_unpin(obj);
-- 
1.7.6

